apps/api/src/schema/lib/year.ts

```diff
@@ -0,0 +1,15 @@
+import { z } from "@hono/zod-openapi";
+
+/**
+ * Expects a string forming a four-digit positive integer
+ */
+
+export const yearSchema = z.coerce
+  .string()
+  .refine((val) => val !== ("" || "undefined" || "null"), {
```

Comment:
Use `required_error` and `invalid_type_error`. This example is given on the Zod docs:
```ts
const name = z.string({
  required_error: "Name is required",
  invalid_type_error: "Name must be a string",
});
```